File: iPad/VerBank-CSTSv3-IOSClientApi/ClientStation/api/info/InfoOperator_TRADESERV1008.m
//
//  InfoOperator_TRADESERV1008.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/8/31.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "InfoOperator_TRADESERV1008.h"

#import "Info_TRADESERV1008.h"
#import "APIDoc.h"
#import "UserDocOperator.h"

@implementation InfoOperator_TRADESERV1008

- (void) onInfo:(InfoFather *)info {
    Info_TRADESERV1008 *tinfo = (Info_TRADESERV1008 *)info;
    if ([tinfo getAttrID] == ATTR_TRANSFER_WITH_FUND) {
        [self infoWithFund:[tinfo getAccountID]];
    }
}

- (void)infoWithFund:(long long)account {
    if ([[APIDoc getUserDocCaptain] getCDS_AccountStore:account] == nil) {
        return;
    }
    
    [[UserDocOperator getInstance] loadMoneyAccount:account];
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/info/InfoOperator_TRADESERV1008.java
package jedi.ex01.client.station.api.info;

import jedi.ex01.CSTS3.comm.info.InfoFather;
import jedi.ex01.CSTS3.comm.info.Info_TRADESERV1008;
import jedi.ex01.client.station.api.doc.DataDoc;
import jedi.ex01.client.station.api.doc.operator.fund.FundDocOperator;
import jedi.ex01.client.station.api.doc.operator.user.UserDocOperator;

public class InfoOperator_TRADESERV1008 extends InfoOperator {

	@Override
	public void onInfo(InfoFather _info) {
		Info_TRADESERV1008 info = (Info_TRADESERV1008) _info;

		if (info.getAttrID() == Info_TRADESERV1008.ATTR_TRANSFER_WITH_FUND) {
			infoWithFund(info.getAccount());
		}
	}

	private void infoWithFund(long account) {
		if (DataDoc.getInstance().getAccountStore().getAccountID() != account) {
			return;
		}
		UserDocOperator.getUserDocOperator().loadMoneyAccount(account);
		FundDocOperator.getFundDocOperator().loadFundAccount(account);
	}

}
